### PR TITLE
Update documentation:

### PR DESCRIPTION
**Pull request by KThompson-Lane**
## Changes:
- Updated the documentation for better clarity and use within our final report
- The various markdown files will be included in the overleaf report as evidence 
- The C_Flat definition markdown file will replace the (now) defunct [google docs ](https://docs.google.com/document/d/1PhXfBH07JUq9n87oGteZDrwJQB-4y3un2L0tGElfcZI) as the single source of truth for the C_Flat definition and EBNF

### CODE REVIEW

File: C_Flat Definition.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+﻿# C_Flat Language Definition
+## Language Syntax:
+
+### Variable types:
+<table>
+    <tr>
+        <th>Token</th>
```
Comment (A-Russell23) [inline C_Flat Definition.md:7]: If these need to match up to our defined token types in TokenType.cs then Number should be Num and Boolean should be String

File: C_Flat Definition.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        <td>Boolean</td>
+        <td>bool</td>
+    </tr>
+</table>
+
+### Operations (Lower precedence evaluates first):
+<table>
+    <tr>
+        <th>Symbol</th>
+        <th>Precedence</th>
```
Comment (A-Russell23) [inline C_Flat Definition.md:28]: All of these should probably have a precedence.

I'm happy to add this

File: C_Flat Definition.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+`<Expression-Query> ::= <Expression> ( '==' | '!=' | '>'| '<' ) <Expression>`
+
+### Conditional Statements:
+
+`<ConditionalStatement>::= 'if’ ‘('<Logic-Statement> ’)’ ‘{' <block> '}' { 'else' ‘{‘ <block> ’}’ }`
+
+`<block>::= *{ <statement> ';' }`
+
+### Variables:
+`<Declaration>::= 'var' <Identifier> '=' (<Expression> | <Logic-Statement> ) ';'`
```
Comment (A-Russell23) [inline C_Flat Definition.md:152]: Declaration, identifier and word are all on the same line. They should all be on their own line

**Review by A-Russell23 — CHANGES_REQUESTED** [review]

File: C_Flat Definition.md (hunk 2 above)
Comment (KThompson-Lane) [reply C_Flat Definition.md:28]: Agreed, ty for taking over on this.

File: C_Flat Definition.md (hunk 3 above)
Comment (KThompson-Lane) [reply C_Flat Definition.md:152]: I think you're mistaken here, declaration is defined on `line 152`, Identifier on `153` and word on `154`

**Review by KThompson-Lane — COMMENTED** [review]
Thanks for making the changes, looks good now.

**Review by A-Russell23 — APPROVED** [review]

**Pull request merged**